addon/components/cf-field/input/table.js

```diff
@@ -62,6 +63,12 @@ export default Component.extend(ComponentQueryManager, {
 
     yield this.onSave(remainingDocuments);
 
+    // Remove orphaned document.
+    yield this.apollo.mutate({
+      mutation: removeDocumentMutation,
+      variables: { input: { document: document.uuid } }
+    });
```

Comment:
I think we need to unload it from the caluma store as well

Reply:
Good catch. Done.